harness/tests/experiment/fixtures/pytorch_onevar_model.py

```diff
@@ -221,13 +227,165 @@ def build_validation_data_loader(self) -> torch.utils.data.DataLoader:
             raise ValueError(f"unknown dataloader_type: {self.hparams['dataloader_type']}")
 
 
-class OneVarTrialWithApexAmp(OneVarTrial):
+class AMPTestDataset(OnesDataset):
+    STAGES = {
+        "one": 1.0,
+        "zero": 0.0,
+        "small": 2e-14,
+        "large": 2e4,
+    }
+
+    def __init__(self, stages: Iterable[str]) -> None:
+        self.stages = stages
+
+    def __len__(self) -> int:
+        return len(self.stages)
+
+    def _get_stage(self, index: int) -> str:
+        stage = self.stages[index]
+        return stage
+
+    def __getitem__(self, index: int) -> Tuple:
+        stage = self._get_stage(index)
+        return self._get_stage_item(stage)
+
+    def _get_stage_item(self, stage) -> Tuple:
+        for _stage, x in self.STAGES.items():
+            if _stage.lower() == stage.lower():
+                y = x
+                return torch.Tensor([float(x)]), torch.Tensor([float(y)])
+        else:
+            raise ValueError(f"Unrecognized {self.__class__.__name__} stage {stage}")
+
+
+class OneVarAMPBaseTrial(OneVarTrial):
+    _dataset_cls = AMPTestDataset
+    _dataset_args = (
+        (
+            5 * ["one"]
+            + 1 * ["large"]
+            + 4 * ["one"]
+            + 1 * ["small"]
+            + 4 * ["one"]
+            + 1 * ["zero"]
+            + 4 * ["one"]
+            + []
+        ),
+    )
+    _init_scale = None
+    _growth_interval = None
+
+    def __init__(self, context: pytorch.PyTorchTrialContext) -> None:
+        context._per_slot_batch_size, context._global_batch_size = 1, 1
```

Comment:
these could potentially be passed into the `make_trial_controller_from_trial_implementation` test method. would avoid setting technically protected variables on the context object. but this is just a test, so not a big deal.

Reply:
Good call. This is done.